server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -231,17 +231,18 @@ private AccessDecision checkAuthorization(RequestDetails requestDetails) {
       return unauthenticatedQueriesDecision;
     }
     // Check the Bearer token to be a valid JWT with required claims.
+
+    AccessDecision allowedQueriesDecision = allowedQueriesChecker.checkAccess(requestDetailsReader);
```

Comment:
The idea for having this after the JWT check was that for "allowed queries", still authentication/authorization is needed. Doesn't this fit your security/access model?

Reply:
@bashir2 related to this - from the code looks like [this field](https://github.com/google/fhir-gateway/blob/4392c87cde108b0a9ca7569576aa5b55aa847d3b/server/src/main/java/com/google/fhir/gateway/AllowedQueriesConfig.java#L46) if set to `true` in the `resources/hapi_page_url_allowed_queries.json` file skips any type of filtering by the proxy - I'm guessing when there's no criteria whatsoever to check for by the proxy for an endpoint. 

I think we need some documentation for this to make it clear for the users.

Reply:
Just a note though, as it is, it may not work for @rehammuzzamil's stated case because if we use it to skip auth. it will be a whitelisting of the whole endpoint/resource rather than via a selective criteria as provided by the allowed queries config.

Reply:
For context though, our requirements are such that we have 3 types of filters:
1. endpoints that should not be authenticated or authorized
2. endpoints that should be authenticated but not authorized
3. endpoints that should be authenticated and authorized

Please note, the authorization I mean here is the customized data sync strategy _e.g. access by location, access by team_ . Similar to the allowed queries we use [our custom configuration file here](https://github.com/google/fhir-gateway/blob/941c4014506d3eb3b39d4da7aece936646f32037/resources/hapi_sync_filter_ignored_queries.json) to enforce this for the specific endpoints.

Reply:
So as discussed off-PR, I think case 2 above can be moved to the custom `AccessChecker` plugin.